src/data/finsupp/basic.lean

```diff
@@ -680,6 +684,9 @@ end
 @[simp] lemma erase_zero (a : α) : erase a (0 : α →₀ M) = 0 :=
 by rw [← support_eq_empty, support_erase, support_zero, erase_empty]
 
+lemma subset_of_support_erase {f : α →₀ M} {q : α}  : (erase q f).support ⊆ f.support :=
+by { simp only [subset_iff, support_erase, mem_erase, ne.def], exact λ a ha, ha.2 }
+
```

Comment:
I don't think this lemma is useful, it follows obviously from
```
by simp only [support_erase, erase_subset]
```
Perhaps `erase_subset` should be a `simp` lemma?

Reply:
This was prompted by a proof where I was left with the goal `(hp: p ∈ (erase q f).support) ⊢ p ∈ f.support`, which I can close with `exact subset_of_support_erase hp`.  I can't see how to do it so briefly otherwise: the best I can do so far is  ```
rw support_erase at hp, exact (finset.erase_subset q f.support) hp
```
(although I'm sure there's a better way that I'm missing).

Anyway, it seemed like the sort of little obvious fact that Lean ought to know about and is easier to find if it has a name.

Reply:
My point is, that if we have simp lemmas that turn `foo f` into `bar f`, we should not be in the business of writing lemmas about both `foo f` and `bar f`, because that doesn't scale at all. If you just write the lemma about `bar f` then the user can still find it with `simp?, library_search`.

Reply:
Your
```
rw support_erase at hp,
exact (finset.erase_subset q f.support) hp
```
looks perfectly reasonable to me. I imagine you should also be able to do `apply finset.erase_subset` if you want a more discoverable proof.

Reply:
Ah, I see what you mean, thanks!